src/spec/gocli/integration/cli_vms_spec.rb

```diff
@@ -101,4 +101,22 @@
     expect(first_row).to include('persistent_disk_usage')
     expect(output.length).to eq(3)
   end
+
+  it "retrieves vms from deployments in parallel" do
+    manifest1 = Bosh::Spec::Deployments.simple_manifest
+    manifest1['name'] = 'simple1'
+    manifest1['jobs'] = [Bosh::Spec::Deployments.simple_job({:name => 'foobar1', :instances => 2})]
+    deploy_from_scratch(manifest_hash: manifest1)
+
+    manifest2 = Bosh::Spec::Deployments.simple_manifest
+    manifest2['name'] = 'simple2'
+    manifest2['jobs'] = [Bosh::Spec::Deployments.simple_job({:name => 'foobar2', :instances => 4})]
+    deploy_from_scratch(manifest_hash: manifest2)
+
+    output = bosh_runner.run('vms --parallel 5')
+    expect(output).to include('Succeeded')
+    # 2 deployments must not be mixed in the output, but they can be printed in any order
+    expect(output).to match(/Deployment 'simple1'\s*\n\nInstance\s+Process\s+State\s+AZ\s+IPs\s+VM\s+CID\s+VM\s+Type\s*\n(foobar1\/\w+-\w+-\w+-\w+-\w+\s+running\s+-\s+\d+.\d+.\d+.\d+\s+\d+\s+a\s*\n){2}\n2 vms/)
+    expect(output).to match(/Deployment 'simple2'\s*\n\nInstance\s+Process\s+State\s+AZ\s+IPs\s+VM\s+CID\s+VM\s+Type\s*\n(foobar2\/\w+-\w+-\w+-\w+-\w+\s+running\s+-\s+\d+.\d+.\d+.\d+\s+\d+\s+a\s*\n){4}\n4 vms/)
```

Comment:
This doesn't really test that parallel checks happen, but I see the value of ensuring results are not interleaved.